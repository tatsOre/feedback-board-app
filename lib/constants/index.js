export const ROADMAP_SECTION_TAGLINES = {
  planned: 'Ideas priorized for research',
  'in-progress': 'Features currently being developed',
  live: 'Released features',
}

export const FILTER_OPTIONS = [
  { label: 'All', value: 'all' },
  { label: 'UI', value: 'ui' },
  { label: 'UX', value: 'ux' },
  { label: 'Enhancement', value: 'enhancement' },
  { label: 'Bug', value: 'bug' },
  { label: 'Feature', value: 'feature' },
]

export const SORT_OPTIONS = [
  { label: 'Most Upvotes', value: 'MOST_UPVOTES' },
  { label: 'Least Upvotes', value: 'LEAST_UPVOTES' },
  { label: 'Most Comments', value: 'MOST_COMMENTS' },
  { label: 'Least Comments', value: 'LEAST_COMMENTS' },
]

export const COLOR_THEME = {
  primary: {
    base: 'bg-violet-900',
    hover: 'hover:bg-violet-500',
    disabled: 'bg-violet-500',
  },
  secondary: { base: 'bg-indigo-800', hover: 'hover:bg-indigo-500' },
  tertiary: { base: 'bg-blue-900', hover: 'hover:bg-blue-500' },
  info: { base: 'bg-blue-500', hover: 'hover:bg-blue-900' },
  danger: { base: 'bg-red-900', hover: 'hover:bg-red-500' },
}

export const CATEGORY_OPTIONS = [
  { label: 'Feature', value: 'feature' },
  { label: 'UI', value: 'ui' },
  { label: 'UX', value: 'ux' },
  { label: 'Enhancement', value: 'enhancement' },
  { label: 'Bug', value: 'bug' },
]

export const STATUS_OPTIONS = [
  { label: 'Suggestion', value: 'suggestion' },
  { label: 'Planned', value: 'planned' },
  { label: 'In-Progress', value: 'in-progress' },
  { label: 'Live', value: 'live' },
]
