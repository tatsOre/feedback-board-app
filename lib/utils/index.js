export const toCapitalize = (text) => {
  if (!text || typeof text !== 'string') return ''
  let label = text
  if (text.includes('-')) label = text.replace('-', ' ')
  return label
    .split(' ')
    .map((w) => w.charAt(0).toUpperCase() + w.slice(1))
    .join(' ')
}

export const getCommentsLength = (comments) => {
  if (!comments || !comments.length) return 0
  return comments.reduce((acc, comment) => {
    return acc + (comment.replies?.length || 0)
  }, comments.length || 0)
}

export const getRandomNumber = (min, max) => {
  let difference = max - min

  let rand = Math.random()

  rand = Math.floor(rand * difference)

  return rand + min
}

export const reduceFeedbacksData = (data) => {
  const accSchema = {
    planned: [],
    'in-progress': [],
    live: [],
    suggestion: [],
  }

  const result = data.reduce((acc, req) => {
    let status = req.status
    let doc = {
      id: req._id.toString(),
      title: req.title,
      slug: req.slug,
      description: req.description,
      status: req.status,
      category: req.category,
      comments: getCommentsLength(req.comments),
      upvotes: req.upvotes,
    }
    acc[status] = acc[status].length ? [...acc[status], doc] : [doc]
    return acc
  }, accSchema)

  return result
}
