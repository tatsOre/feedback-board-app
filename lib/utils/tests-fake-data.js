import MOCK_FEEDBACKS_DATA from '../../__mocks__/mock-feedbacks.json'
import MOCK_USER_DATA from '../../__mocks__/mock-user-info.json'
import MOCK_USER_FEEDBACK_DATA from '../../__mocks__/mock-user-feedback.json'
import MOCK_DETAILED_FEEDBACK_DATA from '../../__mocks__/mock-detailed-feedbacks.json'


export const MOCK_USER = MOCK_USER_DATA.data

export const MOCK_USER_FEEDBACK = MOCK_USER_FEEDBACK_DATA.data

export const MOCK_FEEDBACKS = MOCK_FEEDBACKS_DATA.data

export const MOCK_PLANNED = MOCK_FEEDBACKS.planned

export const MOCK_IN_PROGRESS = MOCK_FEEDBACKS['in-progress']

export const MOCK_LIVE = MOCK_FEEDBACKS.live

export const MOCK_SUGGESTION = MOCK_FEEDBACKS.suggestion

export const MOCK_DETAILED_FEEDBACK = MOCK_DETAILED_FEEDBACK_DATA.data


// import '@testing-library/jest-dom'