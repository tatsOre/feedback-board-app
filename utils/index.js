export const toCapitalize = (text) => {
  if (!text || typeof text !== 'string') return ''
  let label = text
  if (text.includes('-')) label = text.replace('-', ' ')
  return label
    .split(' ')
    .map((w) => w.charAt(0).toUpperCase() + w.slice(1))
    .join(' ')
}

export const getCommentsLength = (comments) => {
  if (!comments.length) return 0
  return comments.reduce((acc, comment) => {
    return acc + (comment.replies?.length || 0)
  }, comments.length || 0)
}

export const getRandomNumber = (min, max) => {
  let difference = max - min

  let rand = Math.random()

  rand = Math.floor(rand * difference)

  return rand + min
}
