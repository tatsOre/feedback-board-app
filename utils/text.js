export const toCapitalize = (text) => {
  if (!text || typeof text !== "string") return "";

  let label = text;
  if (text.includes("-")) label = text.replace("-", " ");

  return label
    .split(" ")
    .map((w) => w.charAt(0).toUpperCase() + w.slice(1))
    .join(" ");
};
