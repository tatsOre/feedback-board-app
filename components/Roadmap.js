import { useState } from 'react'

import { ROADMAP_SECTION_TAGLINES } from '../lib/constants'
import { toCapitalize } from '../lib/utils'

import { AddFeedback } from './Link'
import GoBackButton from './Buttons/GoBack'
import RoadmapCard from './RoadmapCard'


export default function Roadmap({ data }) {
  const [activeTab, setActiveTab] = useState('planned')

  return (
    <div className="container lg:px-4 md:pt-14 lg:pt-20">
      <header className="flex items-center justify-between bg-indigo-900 md:rounded-10 p-6 md:p-7">
        <div>
          <GoBackButton light />
          <h1 className="text-lg md:text-2xl text-white">Roadmap</h1>
        </div>
        <AddFeedback />
      </header>
      <main>
        <nav className="roadmap-tags-nav-mobile md:hidden relative">
          {Object.keys(data).map(
            (status) =>
              status !== 'suggestion' && (
                <button
                  className={`${
                    activeTab === status
                      ? `text-indigo-800 border-b-4 ${status}`
                      : 'text-indigo-500'
                  } w-1/3 h-14 box-content font-bold text-small`}
                  type="button"
                  onClick={() => setActiveTab(status)}
                  key={`roapmap-nav-link-${status}`}
                >
                  {toCapitalize(status)} ({data[status].length})
                </button>
              )
          )}
        </nav>

        <section className="roadmap-feedbacks-section flex p-6 md:p-0 md:pt-8 md:space-x-2.5 lg:space-x-4 xl:space-x-7">
          {Object.keys(data).map(
            (status) =>
              status !== 'suggestion' && (
                <div
                  id={`roapmap-section-${status}`}
                  key={`roapmap-section-${status}`}
                  className={`hidden md:block w-full md:w-1/3 ${
                    activeTab === status && 'active-status'
                  }`}
                >
                  <h2 className="text-indigo-800 text-lg md:text-sm lg:text-lg">
                    {toCapitalize(status)} ({data[status].length})
                  </h2>
                  <p className="text-indigo-500 text-small md:text-sm lg:text-base pb-6">
                    {ROADMAP_SECTION_TAGLINES[status]}
                  </p>
                  {data[status].map((item) => (
                    <RoadmapCard key={item._id} data={item} />
                  ))}
                </div>
              )
          )}
        </section>
      </main>
    </div>
  )
}
