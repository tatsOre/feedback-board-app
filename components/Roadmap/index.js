import Link from 'next/link'
import { useState } from 'react'
import { toCapitalize } from '../../utils/index'
import GoBackButton from '../Buttons/GoBack'
import NavLink from '../NavLink'
import RoadmapCard from './roadmap-card'

const tagline = {
  planned: 'Ideas priorized for research',
  'in-progress': 'Features currently being developed',
  live: 'Released features',
}

const RoadmapPage = ({ data }) => {
  const [show, setShow] = useState('planned')

  return (
    <main className="container lg:px-4 md:pt-14 lg:pt-20">
      <header className="flex items-center justify-between text-white bg-indigo-900 md:rounded-10 p-6 md:p-7">
        <div>
          <GoBackButton light />
          <h1 className="text-lg md:text-2xl">Roadmap</h1>
        </div>

        <Link href="/feedback/new" passHref>
          <NavLink label="+ Add Feedback" variant="primary" />
        </Link>
      </header>

      <nav className="roadmap-tags-nav-mobile md:hidden flex w-full relative">
        {Object.keys(data).map(
          (status) =>
            status !== 'suggestion' && (
              <button
                className={`${
                  show === status
                    ? `text-indigo-800 border-b-4 ${status}`
                    : 'text-indigo-500'
                } w-1/3 h-14 box-content font-bold text-small`}
                type="button"
                onClick={() => setShow(status)}
                key={`roapmap-nav-link-${status}`}
              >
                {toCapitalize(status)} ({data[status].length})
              </button>
            )
        )}
      </nav>

      <section className="roadmap-feedbacks-section flex p-6 md:p-0 md:pt-8 md:space-x-2.5 lg:space-x-4 xl:space-x-7">
        {Object.keys(data).map(
          (status) =>
            status !== 'suggestion' && (
              <div
                key={`roapmap-section-${status}`}
                className={`hidden md:block w-full md:w-1/3 ${
                  show === status && 'active-status'
                }`}
              >
                <h2 className="text-indigo-800 text-lg md:text-sm lg:text-lg">
                  {toCapitalize(status)} ({data[status].length})
                </h2>
                <p className="text-indigo-500 text-small md:text-sm lg:text-base pb-6">
                  {tagline[status]}
                </p>
                {data[status].map((feedback) => (
                  <RoadmapCard key={feedback.id} feedback={feedback} />
                ))}
              </div>
            )
        )}
      </section>
    </main>
  )
}

export default RoadmapPage
