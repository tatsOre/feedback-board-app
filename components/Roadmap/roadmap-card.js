import Link from 'next/link'
import UpvoteButton from '../Buttons/Upvote'
import { getCommentsLength, toCapitalize } from '../../utils'

export default function RoadmapCard({ feedback }) {
  const { category, comments, description, slug, status, title, upvotes } =
    feedback
  const commentsLength = getCommentsLength(comments)
  const categoryText = ['ux', 'ui'].includes(category)
    ? category.toUpperCase()
    : toCapitalize(category)

  return (
    <div
      className={`roadmap-feedback-card ${status} rounded-md border-t-6 bg-white p-6 pt-3 mb-4 text-small lg:text-base`}
    >
      <p className={`status text-indigo-500 mb-3`}>{toCapitalize(status)}</p>
      <Link href={`/feedback/detail/${slug}`} passHref>
        <a>
          <h3 className="text-sm lg:text-lg mb-2 text-indigo-800 hover:text-blue-900 hover:cursor-pointer">
            {title}
          </h3>
        </a>
      </Link>
      <p className="text-indigo-500">{description}</p>
      <p className="w-min rounded-10 bg-indigo-300 px-4 py-1.5 mt-2 mb-4 md:mt-3 lg:text-small font-semibold text-blue-900">
        {categoryText}
      </p>
      <UpvoteButton upvoted={false} upvotes={upvotes} variant="inline" />
      <p className="comments float-right mt-2 font-bold lg:text-base text-indigo-800">
        {commentsLength}
      </p>
    </div>
  )
}
