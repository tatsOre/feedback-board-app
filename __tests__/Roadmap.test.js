import { render, screen, within } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import matchMediaPolyfill from 'mq-polyfill'

import Roadmap from '../components/Roadmap'
import { ROADMAP_SECTION_TAGLINES } from '../lib/constants'
import { MOCK_FEEDBACKS as data } from '../lib/utils/tests-fake-data'

describe('<Roadmap />', () => {
  it('renders with proper ui and props', () => {
    const { container } = render(<Roadmap data={data} />)

    expect(container).toMatchSnapshot()

    expect(screen.getByRole('link', { name: /go back/i })).toHaveAttribute(
      'href',
      '/'
    )

    expect(screen.getByRole('link', { name: /add feedback/i })).toHaveAttribute(
      'href',
      '/feedback/new'
    )
  })

  it('renders sections with proper number of cards, headings and taglines', () => {
    render(<Roadmap data={data} />)

    const planned = document.getElementById('roapmap-section-planned')

    expect(
      within(planned).getByRole('heading', { level: 2 })
    ).toHaveTextContent(`Planned (${data.planned.length})`)

    expect(within(planned).getAllByRole('article')).toHaveLength(
      data.planned.length
    )

    expect(
      screen.getByText(ROADMAP_SECTION_TAGLINES['planned'])
    ).toBeInTheDocument()

    const progress = document.getElementById('roapmap-section-in-progress')

    expect(
      within(progress).getByRole('heading', { level: 2 })
    ).toHaveTextContent(`In Progress (${data['in-progress'].length})`)

    expect(within(progress).getAllByRole('article')).toHaveLength(
      data['in-progress'].length
    )

    expect(
      screen.getByText(ROADMAP_SECTION_TAGLINES['in-progress'])
    ).toBeInTheDocument()

    const live = document.getElementById('roapmap-section-live')

    expect(within(live).getByRole('heading', { level: 2 })).toHaveTextContent(
      `Live (${data.live.length})`
    )

    expect(within(live).getAllByRole('article')).toHaveLength(data.live.length)

    expect(
      screen.getByText(ROADMAP_SECTION_TAGLINES['live'])
    ).toBeInTheDocument()
  })
})

describe('<Roadmap /> with responsive view', () => {
  beforeAll(() => {
    matchMediaPolyfill(window)

    window.resizeTo = function resizeTo(width, height) {
      Object.assign(this, {
        innerWidth: width,
        innerHeight: height,
        outerWidth: width,
        outerHeight: height,
      }).dispatchEvent(new this.Event('resize'))
    }
  })

  it('renders section with "Planned" tab as active, and changes when the user clicks on the tabs', async () => {
    window.resizeTo(500, 1000)

    render(<Roadmap data={data} />)

    expect(screen.getByRole('navigation')).toBeInTheDocument()

    let active = document.getElementsByClassName('active-status')

    expect(active).toHaveLength(1)

    expect(
      within(active[0]).getByRole('heading', { level: 2 })
    ).toHaveTextContent(/planned/i)

    const progressTabButton = screen.getByRole('button', {
      name: /in progress/i,
    })

    await userEvent.click(progressTabButton)

    active = document.getElementsByClassName('active-status')

    expect(active).toHaveLength(1)

    expect(
      within(active[0]).getByRole('heading', { level: 2 })
    ).toHaveTextContent(/in progress/i)

    const liveTabButton = screen.getByRole('button', {
      name: /live/i,
    })

    await userEvent.click(liveTabButton)

    active = document.getElementsByClassName('active-status')

    expect(active).toHaveLength(1)

    expect(
      within(active[0]).getByRole('heading', { level: 2 })
    ).toHaveTextContent(/live/i)
  })
})
