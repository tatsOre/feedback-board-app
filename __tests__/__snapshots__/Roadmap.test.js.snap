// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Roadmap /> renders with proper ui and props 1`] = `
<div>
  <div
    class="container lg:px-4 md:pt-14 lg:pt-20"
  >
    <header
      class="flex items-center justify-between bg-indigo-900 md:rounded-10 p-6 md:p-7"
    >
      <div>
        <a
          class="text-sm hover:underline font-bold text-white"
          href="/"
        >
          <svg
            class="mb-1 mr-3"
            height="10"
            width="7"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 9L2 5l4-4"
              fill="none"
              fill-rule="evenodd"
              stroke="#FFFFFF"
              stroke-width="2"
            />
          </svg>
          Go Back
        </a>
        <h1
          class="text-lg md:text-2xl text-white"
        >
          Roadmap
        </h1>
      </div>
      <a
        class="bg-violet-900 hover:bg-violet-500 text-white text-[13px] md:text-sm font-bold px-6 py-2.5 rounded-10 inline-block min-w-fit"
        href="/feedback/new"
      >
        + Add Feedback
      </a>
    </header>
    <main>
      <nav
        class="roadmap-tags-nav-mobile md:hidden relative"
      >
        <button
          class="text-indigo-800 border-b-4 planned w-1/3 h-14 box-content font-bold text-small"
          type="button"
        >
          Planned
           (
          2
          )
        </button>
        <button
          class="text-indigo-500 w-1/3 h-14 box-content font-bold text-small"
          type="button"
        >
          In Progress
           (
          3
          )
        </button>
        <button
          class="text-indigo-500 w-1/3 h-14 box-content font-bold text-small"
          type="button"
        >
          Live
           (
          1
          )
        </button>
      </nav>
      <section
        class="roadmap-feedbacks-section flex p-6 md:p-0 md:pt-8 md:space-x-2.5 lg:space-x-4 xl:space-x-7"
      >
        <div
          class="hidden md:block w-full md:w-1/3 active-status"
          id="roadmap-section-planned"
        >
          <h2
            class="text-indigo-800 text-lg md:text-sm lg:text-lg"
          >
            Planned
             (
            2
            )
          </h2>
          <p
            class="text-indigo-500 text-small md:text-sm lg:text-base pb-6"
          >
            Ideas priorized for research
          </p>
          <article
            class="roadmap-feedback-card planned rounded-md border-t-6 bg-white p-6 pt-3 mb-4 text-small lg:text-base"
          >
            <p
              class="status text-indigo-500 mb-3"
            >
              Planned
            </p>
            <h3
              class="text-sm lg:text-lg mb-2"
            >
              <a
                class="text-indigo-800 hover:text-blue-900 hover:cursor-pointer"
                href="/feedback/more-comprehensive-reports/detail"
              >
                More comprehensive reports
              </a>
            </h3>
            <p
              class="text-indigo-500"
            >
              It would be great to see a more detailed breakdown of solutions.
            </p>
            <p
              class="w-min rounded-10 bg-indigo-300 px-4 py-1.5 mt-2 mb-4 md:mt-3 lg:text-small font-semibold text-blue-900"
            >
              Feature
            </p>
            <button
              aria-label="upvote"
              class="bg-indigo-300 text-indigo-800 hover:bg-indigo-400 rounded-10 text-[13px] min-w-[70px] py-3 px-4 leading-3 flex-inline justify-evenly relative z-10"
              type="button"
            >
              <svg
                height="7"
                width="10"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 6l4-4 4 4"
                  fill="none"
                  fill-rule="evenodd"
                  stroke="#4661E6"
                  stroke-width="2"
                />
              </svg>
              123
            </button>
            <p
              aria-label="comments"
              class="text-indigo-800 comments float-right mt-2 font-bold lg:text-base text-indigo-800"
            >
              2
            </p>
          </article>
          <article
            class="roadmap-feedback-card planned rounded-md border-t-6 bg-white p-6 pt-3 mb-4 text-small lg:text-base"
          >
            <p
              class="status text-indigo-500 mb-3"
            >
              Planned
            </p>
            <h3
              class="text-sm lg:text-lg mb-2"
            >
              <a
                class="text-indigo-800 hover:text-blue-900 hover:cursor-pointer"
                href="/feedback/learning-paths/detail"
              >
                Learning paths
              </a>
            </h3>
            <p
              class="text-indigo-500"
            >
              Sequenced projects for different goals to help people improve.
            </p>
            <p
              class="w-min rounded-10 bg-indigo-300 px-4 py-1.5 mt-2 mb-4 md:mt-3 lg:text-small font-semibold text-blue-900"
            >
              Feature
            </p>
            <button
              aria-label="upvote"
              class="bg-indigo-300 text-indigo-800 hover:bg-indigo-400 rounded-10 text-[13px] min-w-[70px] py-3 px-4 leading-3 flex-inline justify-evenly relative z-10"
              type="button"
            >
              <svg
                height="7"
                width="10"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 6l4-4 4 4"
                  fill="none"
                  fill-rule="evenodd"
                  stroke="#4661E6"
                  stroke-width="2"
                />
              </svg>
              28
            </button>
            <p
              aria-label="comments"
              class="text-indigo-800 comments float-right mt-2 font-bold lg:text-base text-indigo-800"
            >
              1
            </p>
          </article>
        </div>
        <div
          class="hidden md:block w-full md:w-1/3 false"
          id="roadmap-section-in-progress"
        >
          <h2
            class="text-indigo-800 text-lg md:text-sm lg:text-lg"
          >
            In Progress
             (
            3
            )
          </h2>
          <p
            class="text-indigo-500 text-small md:text-sm lg:text-base pb-6"
          >
            Features currently being developed
          </p>
          <article
            class="roadmap-feedback-card in-progress rounded-md border-t-6 bg-white p-6 pt-3 mb-4 text-small lg:text-base"
          >
            <p
              class="status text-indigo-500 mb-3"
            >
              In Progress
            </p>
            <h3
              class="text-sm lg:text-lg mb-2"
            >
              <a
                class="text-indigo-800 hover:text-blue-900 hover:cursor-pointer"
                href="/feedback/one-click-portfolio-generation/detail"
              >
                One-click portfolio generation
              </a>
            </h3>
            <p
              class="text-indigo-500"
            >
              Add ability to create professional looking portfolio from profile.
            </p>
            <p
              class="w-min rounded-10 bg-indigo-300 px-4 py-1.5 mt-2 mb-4 md:mt-3 lg:text-small font-semibold text-blue-900"
            >
              Feature
            </p>
            <button
              aria-label="upvote"
              class="bg-indigo-300 text-indigo-800 hover:bg-indigo-400 rounded-10 text-[13px] min-w-[70px] py-3 px-4 leading-3 flex-inline justify-evenly relative z-10"
              type="button"
            >
              <svg
                height="7"
                width="10"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 6l4-4 4 4"
                  fill="none"
                  fill-rule="evenodd"
                  stroke="#4661E6"
                  stroke-width="2"
                />
              </svg>
              62
            </button>
            <p
              aria-label="comments"
              class="text-indigo-800 comments float-right mt-2 font-bold lg:text-base text-indigo-800"
            >
              1
            </p>
          </article>
          <article
            class="roadmap-feedback-card in-progress rounded-md border-t-6 bg-white p-6 pt-3 mb-4 text-small lg:text-base"
          >
            <p
              class="status text-indigo-500 mb-3"
            >
              In Progress
            </p>
            <h3
              class="text-sm lg:text-lg mb-2"
            >
              <a
                class="text-indigo-800 hover:text-blue-900 hover:cursor-pointer"
                href="/feedback/bookmark-challenges/detail"
              >
                Bookmark challenges
              </a>
            </h3>
            <p
              class="text-indigo-500"
            >
              Be able to bookmark challenges to take later on.
            </p>
            <p
              class="w-min rounded-10 bg-indigo-300 px-4 py-1.5 mt-2 mb-4 md:mt-3 lg:text-small font-semibold text-blue-900"
            >
              Feature
            </p>
            <button
              aria-label="upvote"
              class="bg-indigo-300 text-indigo-800 hover:bg-indigo-400 rounded-10 text-[13px] min-w-[70px] py-3 px-4 leading-3 flex-inline justify-evenly relative z-10"
              type="button"
            >
              <svg
                height="7"
                width="10"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 6l4-4 4 4"
                  fill="none"
                  fill-rule="evenodd"
                  stroke="#4661E6"
                  stroke-width="2"
                />
              </svg>
              31
            </button>
            <p
              aria-label="comments"
              class="text-indigo-800 comments float-right mt-2 font-bold lg:text-base text-indigo-800"
            >
              1
            </p>
          </article>
          <article
            class="roadmap-feedback-card in-progress rounded-md border-t-6 bg-white p-6 pt-3 mb-4 text-small lg:text-base"
          >
            <p
              class="status text-indigo-500 mb-3"
            >
              In Progress
            </p>
            <h3
              class="text-sm lg:text-lg mb-2"
            >
              <a
                class="text-indigo-800 hover:text-blue-900 hover:cursor-pointer"
                href="/feedback/animated-solution-screenshots/detail"
              >
                Animated solution screenshots
              </a>
            </h3>
            <p
              class="text-indigo-500"
            >
              Screenshots of solutions with animations don’t display correctly.
            </p>
            <p
              class="w-min rounded-10 bg-indigo-300 px-4 py-1.5 mt-2 mb-4 md:mt-3 lg:text-small font-semibold text-blue-900"
            >
              Bug
            </p>
            <button
              aria-label="upvote"
              class="bg-indigo-300 text-indigo-800 hover:bg-indigo-400 rounded-10 text-[13px] min-w-[70px] py-3 px-4 leading-3 flex-inline justify-evenly relative z-10"
              type="button"
            >
              <svg
                height="7"
                width="10"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 6l4-4 4 4"
                  fill="none"
                  fill-rule="evenodd"
                  stroke="#4661E6"
                  stroke-width="2"
                />
              </svg>
              9
            </button>
            <p
              aria-label="comments"
              class="text-[#898EAC] comments float-right mt-2 font-bold lg:text-base text-indigo-800"
            >
              0
            </p>
          </article>
        </div>
        <div
          class="hidden md:block w-full md:w-1/3 false"
          id="roadmap-section-live"
        >
          <h2
            class="text-indigo-800 text-lg md:text-sm lg:text-lg"
          >
            Live
             (
            1
            )
          </h2>
          <p
            class="text-indigo-500 text-small md:text-sm lg:text-base pb-6"
          >
            Released features
          </p>
          <article
            class="roadmap-feedback-card live rounded-md border-t-6 bg-white p-6 pt-3 mb-4 text-small lg:text-base"
          >
            <p
              class="status text-indigo-500 mb-3"
            >
              Live
            </p>
            <h3
              class="text-sm lg:text-lg mb-2"
            >
              <a
                class="text-indigo-800 hover:text-blue-900 hover:cursor-pointer"
                href="/feedback/add-micro-interactions/detail"
              >
                Add micro-interactions
              </a>
            </h3>
            <p
              class="text-indigo-500"
            >
              Small animations at specific points can add delight.
            </p>
            <p
              class="w-min rounded-10 bg-indigo-300 px-4 py-1.5 mt-2 mb-4 md:mt-3 lg:text-small font-semibold text-blue-900"
            >
              Enhancement
            </p>
            <button
              aria-label="upvote"
              class="bg-indigo-300 text-indigo-800 hover:bg-indigo-400 rounded-10 text-[13px] min-w-[70px] py-3 px-4 leading-3 flex-inline justify-evenly relative z-10"
              type="button"
            >
              <svg
                height="7"
                width="10"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 6l4-4 4 4"
                  fill="none"
                  fill-rule="evenodd"
                  stroke="#4661E6"
                  stroke-width="2"
                />
              </svg>
              71
            </button>
            <p
              aria-label="comments"
              class="text-indigo-800 comments float-right mt-2 font-bold lg:text-base text-indigo-800"
            >
              2
            </p>
          </article>
        </div>
      </section>
    </main>
  </div>
</div>
`;
